web/profiles/joinup/src/Plugin/Block/AddContentBlock.php

```diff
@@ -98,25 +98,27 @@ public function build() {
         'rdf_entity' => $collection_contexts['collection']->getContextValue()
           ->id(),
       ]);
-      $links['custom_page'] = [
-        '#type' => 'link',
-        '#title' => $this->t('Add custom page'),
-        '#url' => $page_url,
-        '#attributes' => ['class' => ['button', 'button--small']],
-        '#access' => $page_url->access(),
-      ];
+      if ($page_url->access()) {
+        $links['custom_page'] = [
+          '#type' => 'link',
+          '#title' => $this->t('Add custom page'),
+          '#url' => $page_url,
+          '#attributes' => ['class' => ['button', 'button--small']],
+        ];
+      }
```

Comment:
Yup! There are actually two tickets that are marked as fulfilled by this ticket :)
This is indeed a nice one. I need to start using these inline functions though I must admit that since they are not that popular in terms of usage, it might not be that easy for newcomers to quickly adapt! but I like it! I will change it.